stories/components/table.stories.tsx

```diff
@@ -187,3 +194,146 @@ export function StyledTable(props: ControlProps) {
     />
   );
 }
+
+export const ScrollToVirtualRow = {
+  args: {
+    scrollBehavior: 'auto',
+    scrollAlign: 'start',
+    stickyHeader: true,
+  },
+  argTypes: {
+    scrollBehavior: {
+      control: {
+        type: 'select',
+      },
+      options: ['auto', 'smooth'],
+    },
+    scrollAlign: {
+      control: {
+        type: 'select',
+      },
+      options: ['auto', 'center', 'start', 'end'],
+    },
+  },
+  render: (props) => {
+    const { rowIndex, buttons } = useScrollButtons();
+    const scrollToRef = useRef<VirtualScrollToRow>();
+    useEffect(() => {
+      if (scrollToRef.current) {
+        scrollToRef.current(String(rowIndex), {
+          align: props.scrollAlign,
+          behavior: props.scrollBehavior,
+        });
+      }
+    }, [rowIndex, props.scrollAlign, props.scrollBehavior]);
+
+    return (
+      <div style={{ display: 'flex', flexDirection: 'column', height: '100%' }}>
+        {buttons}
+        <div style={{ flex: 1, overflow: 'auto', minHeight: 0 }}>
+          <Table
+            {...props}
+            virtualizeRows
+            data={table}
+            tableProps={{
+              style: { height: '100%' },
+            }}
+            columns={columns}
+            estimatedRowHeight={() => 172}
+            scrollToRowRef={scrollToRef}
+          />
+        </div>
+      </div>
+    );
+  },
+} satisfies Meta;
+
+export const ScrollRowIntoView = {
+  args: {
+    scrollBehavior: 'instant',
+    scrollBlock: 'start',
+    stickyHeader: false,
+  },
+  argTypes: {
+    scrollBehavior: {
+      control: {
+        type: 'select',
+      },
+      options: ['auto', 'smooth', 'instant'],
+    },
+    scrollBlock: {
+      control: {
+        type: 'select',
+      },
+      options: ['nearest', 'center', 'start', 'end'],
+    },
+    virtualizeRows: {
+      table: {
+        disable: true,
+      },
+    },
+  },
+  render: (props) => {
+    const { rowIndex, buttons } = useScrollButtons();
+    const scrollToRef = useRef<ScrollToRow>();
+    useEffect(() => {
+      if (scrollToRef.current) {
+        scrollToRef.current(String(rowIndex), {
+          behavior: props.scrollBehavior,
+          block: props.scrollBlock,
+        });
+      }
+    }, [rowIndex, props.scrollBehavior, props.scrollBlock]);
```

Comment:
Can you make a more "React correct" example? You're not supposed to use an effect to do something in reaction to a user clicking a button. Just do the thing in the click handler.

Reply:
Ok, I'll change. I think I've done it that way intuitively because there are situations where it is required that scrolling happens after a state update is rendered on the screen, for example following an action that changed the data table.